### PR TITLE
jsm: Add StreamNames and ConsumerNames

### PR DESCRIPTION
**Pull request by variadico**
This adds `ConsumerNames` and `StreamNames` to `JetStreamManager`.

```go
StreamNames(ctx context.Context) <-chan string
ConsumerNames(ctx context.Context, stream string) <-chan string
```

### CODE REVIEW

File: test/js_test.go [hunk 1]
```diff
@@ -697,129 +697,164 @@ func TestJetStreamManagement(t *testing.T) {
...
-	// Delete a consumer using our client API.
-	if err := js.DeleteConsumer("", ""); err == nil {
-		t.Fatalf("Unexpected success")
-	}
-	if err := js.DeleteConsumer("foo", "dlc"); err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
+	t.Run("list consumer names", func(t *testing.T) {
+		var names []string
+		for name := range js.ConsumerNames(context.Background(), "foo") {
```
Comment (wallyqs) [inline test/js_test.go:809]: context.WithTimeout in case it could block?

File: test/js_test.go [hunk 2]
```diff
@@ -697,129 +697,164 @@ func TestJetStreamManagement(t *testing.T) {
...
+		if si, err := js.StreamInfo("foo"); err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		} else if si.State.Msgs != 0 {
+			t.Fatalf("StreamInfo.Msgs is not correct")
+		}
+	})
+
+	t.Run("list stream names", func(t *testing.T) {
+		var names []string
+		for name := range js.StreamNames(context.Background()) {
```
Comment (wallyqs) [inline test/js_test.go:839]: add nil or timeout so that it cannot block

File: jsm.go [hunk 3]
```diff
@@ -679,3 +778,90 @@ func (s *StreamLister) Err() error {
...
+func (l *streamNamesLister) Next() bool {
+	if l.err != nil {
+		return false
+	}
+	if l.pageInfo != nil && l.offset >= l.pageInfo.Total {
+		return false
+	}
+
+	subj := l.js.apiSubj(fmt.Sprintf(apiStreamNamesT))
+	r, err := l.js.nc.Request(subj, nil, l.js.wait)
```
Comment (wallyqs) [inline jsm.go:808]: Seems this is the same request as this one but with an empty payload? https://github.com/nats-io/nats.go/blob/master/js.go#L655-L676

File: js.go [hunk 4]
```diff
@@ -73,6 +76,9 @@ const (
 	// apiStreamListT is the endpoint that will return all detailed stream information
 	apiStreamList = "STREAM.LIST"
 
+	// apiStreamNamesT is used to return a list with all stream names.
+	apiStreamNamesT = "STREAM.NAMES"
```
Comment (wallyqs) [inline js.go:80]: This is the same as `apiStreams = "STREAM.NAMES"` which above

File: jsm.go (hunk 3 above)
Comment (variadico) [reply jsm.go:808]: Good catch. I deleted `apiStreamNamesT` and `streamNamesListResponse`.

**Comment by wallyqs** [comment]
Let's merge this one after rebase, should be ok since this does not replace APIs used by the server.

**Comment by variadico** [comment]
Rebased. 👍🏽

**Comment by wallyqs** [comment]
ah we also need squash

**Pull request merged**